### PR TITLE
feat: ✨ add nestedStackedBarsVertical and confarea

### PR DESCRIPTION
**Pull request by lloydrichards**
## Goals/Scope

We have identified two opportunities to improve the SSZVIS library by incorporating commonly used features, the `nestedStackedBar` and the `confarea`. Here are the key points:

- Currently, these features are implemented across 35 charts (5% of all charts, more than `heatmap`, `sunburst` and `pie chart` combined).
- The widespread use indicates its importance and utility in our visualisations (11% of line charts include `confarea`).
- Maintaining separate implementations increases the risk of bugs and complicates future maintenance efforts.

### Proposed Solution

To address this, we propose:

- Integrating the existing `nestedStackedBar` and `confarea` implementation directly into the SSZVIS library (bump version to 3.1.0).
- Removing individual implementations from specific charts reduces the overall code surface area.
- Add documentation on how to use and build charts using these features

### Additional Tasks

There are a few underlying issues with charts that are related to the old .enter().exit() pattern. We will address these issues as part of this PR by updating the code to use the new .join() pattern.

## Description

Porting the `nestedStackedBar` and `confarea` features went as smooth as expected. There is now a new page for the nestedStackedBar and confarea was added to the annotation section:

### `sszvis.nestedStackedBarsVertical`

- [x] Ported the nestedStackedBar feature to the SSZVIS library
- [x] Added documentation to a new page on how to use the nestedStackedBar feature
- [x] Added example chart

### `sszvis.annotationConfArea`

- [x] Ported the confarea feature to the SSZVIS library
- [x] Added documentation to the Annotation page on how to use the confarea feature
- [x] Added example chart

With the `.join()` pattern there was actually a lot of places this needed to be changes and many side effects that needed to be addressed. `.enter()` has now been removed from the codebase and all the charts tested for visual regression and functionality.

Will be good to do some more testing with all the production charts to make sure everything is working as expected.

### How to Test

In order to see the charts you will need to spin up the local server and navigate to the nestedStackedBar and Annotation pages. (make sure to run `npm run build` before starting the server)

---

- [x] Timeframe: [`24 hr`]
- [x] Urgent!

### CODE REVIEW

File: src/annotation/ruler.js [hunk 1]
```diff
@@ -180,4 +173,55 @@ export default function () {
...
+    labelBounds.push({
+      category: cAcc(d),
+      top: bounds.top,
+      bottom: bounds.bottom,
+      dy: 0,
+    });
+  });
+
+  // Sort by vertical position (only supports labels of same height)
+  labelBounds = labelBounds.sort((a, b) => ascending(a.top, b.top));
```
Comment (wereHamster) [inline src/annotation/ruler.js:198]:
```suggestion
  labelBounds = labelBounds.sort((a, b) => ascending(a.top, b.top));
```

sort function mutates then array in-place, no need to assign.

File: src/annotation/ruler.js [hunk 2]
```diff
@@ -180,4 +173,55 @@ export default function () {
         });
       }
     });
-}
+
+export const rulerLabelVerticalSeparate = (cAcc) => (g) => {
+  const THRESHOLD = 2;
+  let labelBounds = [];
```
Comment (wereHamster) [inline src/annotation/ruler.js:179]: can be const

File: src/component/nestedStackedBar.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      const selection = select(this);
+      const props = selection.props();
+
+      const offset = props.offset;
+      const xScale = props.xScale;
+      const yScale = props.yScale;
+      const fill = props.fill;
+      const tooltip = props.tooltip;
+      const xAcc = props.xAcc;
+      const xLabel = props.xLabel;
```
Comment (wereHamster) [inline src/component/nestedStackedBar.js:56]: Can we use destructuring? I mean in the sense that the browsers we officially support know this syntax

```js
const { offset, xScale, yScale } = props
```

File: docs/guides/annotations.md [hunk 4]
```diff
@@ -299,3 +299,53 @@ A value for the y-value of the upper range flag dot
     "sourceView": ["index.html", "data.csv"]
 }
 ```
+
+## Confidence Area
+
+The area is a shaded area that represents a confidence interval. It is used to show the range of values within which the true value is likely to fall.
+
+### sszvis.annotationConfArea
```
Comment (wereHamster) [inline docs/guides/annotations.md:307]: I'd write this out, ie. `sszvis.annotationConfidenceArea`. The short form `Conf` may be ambigous or difficult to understand for someone not familar with the project.

**Review by wereHamster — APPROVED** [review]
code looks good, I assume you tested with the two examples

File: src/component/nestedStackedBar.js (hunk 3 above)
Comment (lloydrichards) [reply src/component/nestedStackedBar.js:56]: this is a bit of a convention from the other components in the library, but I can change it up for this one now and maybe layer refactor the others

**Pull request merged**